eth/stagedsync/stage_headers.go

```diff
@@ -164,16 +164,14 @@ func HeadersPOS(
 	cfg.hd.SetHeaderReader(&chainReader{config: &cfg.chainConfig, tx: tx, blockReader: cfg.blockReader})
 	headerInserter := headerdownload.NewHeaderInserter(s.LogPrefix(), nil, s.BlockNumber, cfg.blockReader)
 
-	if interrupt != engineapi.None {
-		if interrupt == engineapi.Stopping {
-			cfg.hd.PayloadStatusCh <- privateapi.PayloadStatus{CriticalError: errors.New("server is stopping")}
-		}
-		if interrupt == engineapi.Synced {
-			verifyAndSaveDownloadedPoSHeaders(tx, cfg, headerInserter)
-		}
-		if !useExternalTx {
-			return tx.Commit()
-		}
+	var err error
+	var interrupted bool
+	interrupted, err = handleInterrupt(interrupt, cfg, tx, headerInserter, useExternalTx)
```

Comment:
More concise would be to use the [short variable declaration](https://go.dev/ref/spec#Short_variable_declarations) `:=` [instead of the declaration + definition](https://stackoverflow.com/questions/17891226/difference-between-and-operators-in-go).